File: typescript-api/src/data/services/update-user.ts
import { User } from "../../domain/entities";
import { UpdateUserRepository } from "../repositories";

export class UpdateUserService {
  constructor(
    private readonly repo: UpdateUserRepository
  ) {}

  async update(userId: number, newUser: Omit<User, 'id' | 'password'>): Promise<void> {
    this.repo.update(userId, newUser)
  }
}

File: rust-api/src/data/services/update_user.rs
use crate::data::{ 
    repositories::UpdateUserRepository,
    dto::UpdateUserDto
};
use std::sync::Arc;

pub async fn update_user_service(
    repo: Arc<impl UpdateUserRepository>,
    user_id: i32,
    new_user: &UpdateUserDto,
) -> Result<(), sqlx::Error> {
    repo.update(user_id, new_user).await
}
